File: 3471-MinimumAverageOfSmallestAndLargestElements/3471-MinimumAverageOfSmallestAndLargestElements.java
// Last updated: 31/07/2025, 21:04:47
class Solution {
    public double minimumAverage(int[] nums) {
        Arrays.sort(nums);
        double ma=nums[nums.length-1];
        for(int i=0;i< nums.length / 2;i++)
        {
            double avg=(nums[i]+nums[nums.length-i-1])/2.0;
            if(avg<ma)
            {
                ma=avg;
            }
        }
        return ma;
    }
}

File: 3471-MinimumAverageOfSmallestAndLargestElements/3471-MinimumAverageOfSmallestAndLargestElements.py
# Last updated: 31/07/2025, 21:04:52
class Solution:
    def minimumAverage(self, nums: List[int]) -> float:
        a=[]
        n=len(nums)
        nums.sort()
        for i in range(n//2):
            a.append((nums[0]+nums[-1])/2)
            nums.pop(0)
            nums.pop(-1)
        return min(a)    
